js/form/src/DropdownOptions.svelte

```diff
@@ -10,9 +10,30 @@
 	let distance_from_top: number;
 	let distance_from_bottom: number;
 	let input_height: number;
+	let input_width: number;
 	let refElement: HTMLDivElement;
 	let listElement: HTMLUListElement;
 	let top: string | null, bottom: string | null, max_height: number;
+	let innerHeight: number;
+
+	const calculate_window_distance = () => {
+    const {top, bottom} = refElement.getBoundingClientRect(); 
+		distance_from_top = innerHeight - top;
+		distance_from_bottom = innerHeight - bottom;
+	};
+
+	let scroll_timeout: NodeJS.Timeout | null = null;
+	const scroll_listener = () => {
+		if (scroll_timeout !== null) {
+			clearTimeout(scroll_timeout);
+		}
+
+		scroll_timeout = setTimeout(() => {
+			calculate_window_distance();
+			scroll_timeout = null;
+		}, 200);
```

Comment:
I would change this to 5/ 10 ms or something. This is mainly to ensure we aren't firing thousand of callbacks off per second, we still want to do it quickly to keep things looking good.